### PR TITLE
Front

### PR DESCRIPTION
**Pull request by emmanuel-lbs**
- token half works  (if a do : get user by token at back i dont get the user -> we have to decrypt the token to get id )
- des cookies perfecto (i stock token in cookie and if user reload page i take cookie token and fetch to back to get user
- buttons search on profile page qui fetch get user by username au back but  redirect not yet
- the form to connect is now require the new name but  not necessarily the avatar
- des routes private and public ( if not log the man cant go on other page than / and if log he cant go on / )
- take off header on connexion page 
- change all localhost by var in .env
- change game.services.ts by  game.service.ts
- change details in front game (for the result i put a bool for display a pop up after
- little clean (== change by === and != by !== ...)

### CODE REVIEW

File: front/src/App.tsx [hunk 1]
```diff
@@ -11,12 +15,46 @@ import router from './router';
...
 function App() {
-  const myUser = useAppSelector(state => state.user);
+  const myStore = useAppSelector(state => state.user);
+  const dispatch = useAppDispatch();
+  const cookies = new Cookies();
+  const token = cookies.get('Token');
+
+
+  useEffect(() => {
+    if (myStore.isLog == true) {
```
Comment (Blodispic) [inline front/src/App.tsx:25]: Le if sert a quoi ici ? Tu va faire quelque chose avec ? :)

**Review by Blodispic — APPROVED** [review]
Good good !

File: front/src/App.tsx (hunk 1 above)
Comment (emmanuel-lbs) [reply front/src/App.tsx:25]: C'est pour les socket : Adam s'en occupe ils avais juste besoin de ça

File: back/src/user/user.service.ts [hunk 2]
```diff
@@ -38,6 +38,19 @@ export class UserService {
...
+ async GetByAccessToken(access_token: string) {
+    console.log(access_token);
+     console.log("cadis qioi");
+     const teoken = await this.usersRepository.findOne({
+       where: {
+         access_token: access_token
+       }
+    })
+    console.log(teoken);
+    return teoken;
```
Comment (Acusanno) [inline back/src/user/user.service.ts:50]: Pourquoi "teoken" au lieu de "token"?

Comment (Blodispic) [reply back/src/user/user.service.ts:50]: C'est Manu quoi

**Review by Acusanno — APPROVED** [review]

**Pull request merged**